appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1661,14 +1652,12 @@ private void deployAll() throws GlassFishException {
                     try {
                         // Convert the URL to a URI for use with the deploy method
                         URI artefactURI = deploymentMapEntry.getValue().toURI();
-
-                        String artefactName = getFileName(artefactURI.getPath());
-                        String name = artefactName.substring(0, artefactName.length() - 4);
+                        String artefactName = removeJavaArchiveExtension(new File(artefactURI).getName());
```

Comment:
If the artefactURI does not have the file scheme will we not get IllegalArgumentException?

Reply:
Don’t know. Please test and change as appropriate